packages/base/src/signals/signal-manager.ts

```diff
@@ -26,8 +26,8 @@ export declare interface SignalManager {
     fireTraceServerStartedSignal(): void;
     fireUndoSignal(): void;
     fireRedoSignal(): void;
-    firePinView(output: OutputDescriptor): void;
-    fireUnPinView(): void;
+    firePinView(payload: { output: OutputDescriptor, checkedSeries?: number[], collapsedNodes?: number[], collapsedMarkerNodes?: number[]}): void;
```

Comment:
It feels like the persisted state of outputs should be opaque as it can be different for every output (e.g. there are no checkedSeries in timegraph). It doesn't feel right to have a merge of all possible state properties here.

I would perhaps keep the output and persistedState as separate parameters in the signal. I'm not sure if the persisted state can be an 'any' object or a choice of state classes defined in each specific output.

But I think signal-manager, abstract-output-component and trace-context-component should pass the persisted state as a 'black box' object.

Reply:
Thanks for the suggestion Patrick, I've added another commit which passes the persisted state as a 'black-box' object. I was able to specify the persisted state as an 'any' object with the help of disable es-lint statements.